### PR TITLE
crud: support vshard with master discovery

### LINKED ISSUE #409: support vshard `master: auto`
It was introduced in PR https://github.com/tarantool/vshard/pull/439

### PR DESCRIPTION
**Pull request by DifferentialOrange**
In vshard 0.1.25 and Tarantool 3.0 new feature has been implemented. The feature is related to automatic master discovery. If it has been enabled, crud fails to bootstrap and work in several places. This feature is enabled by Tarantool 3.0 if vshard cluster was configured with specific 3.0 config flags.

Unfortunately, it is impossible to rework the code using only vshard public API now [2].

1. https://github.com/tarantool/vshard/issues/429
2. https://github.com/tarantool/vshard/issues/467

Closes #409

### CODE REVIEW

**Comment by DifferentialOrange** [comment]
Locally, some map call tests had failed for me. I'll explore it after the holidays.
```
1) truncate.backend:"vshard".backend_cfg:{identification_mode = "name_as_key", master = "auto"}.engine:"vinyl".test_non_existent_space
...thub/tarantool/crud///test/integration/truncate_test.lua:31: expected: nil, actual: true
stack traceback:
  ...thub/tarantool/crud///test/integration/truncate_test.lua:31: in function 'truncate.backend:"vshard".backend_cfg:{identification_mode = "name_as_key", master = "auto"}.engine:"vinyl".test_non_existent_space'
  ...
  [C]: in function 'xpcall'
2) truncate.backend:"vshard".backend_cfg:{identification_mode = "uuid_as_key", master = "auto"}.engine:"vinyl".test_non_existent_space
...thub/tarantool/crud///test/integration/truncate_test.lua:31: expected: nil, actual: true
stack traceback:
  ...thub/tarantool/crud///test/integration/truncate_test.lua:31: in function 'truncate.backend:"vshard".backend_cfg:{identification_mode = "uuid_as_key", master = "auto"}.engine:"vinyl".test_non_existent_space'
  ...
  [C]: in function 'xpcall'
Ran 24 tests in 13.557 seconds, 22 succeeded, 2 failed
```

File: crud.lua [hunk 1]
```diff
@@ -173,15 +171,7 @@ function crud.init_storage()
...
-        local replicaset_key, replicaset = utils.get_self_vshard_replicaset()
-
-        if replicaset == nil or replicaset.master == nil then
-            error(string.format(
-                'Failed to find a vshard configuration ' ..
-                'for storage replicaset with key %q.',
-                replicaset_key))
-        end
-        user = luri.parse(replicaset.master.uri).login or 'guest'
+        user = utils.get_this_replica_user() or 'guest'
```
Comment (askalt) [inline crud.lua:174]: `vshard_utils` instead of `utils`?

**Review by better0fdead — APPROVED** [review]
Hi, thx for patch

**Review by askalt — APPROVED** [review]
Thank you for the patch!

**Pull request merged**